homeassistant/components/generic_thermostat/climate.py

```diff
@@ -82,6 +89,22 @@
             [HVAC_MODE_COOL, HVAC_MODE_HEAT, HVAC_MODE_OFF]
         ),
         vol.Optional(CONF_AWAY_TEMP): vol.Coerce(float),
+        vol.Optional(CONF_PRESETS): vol.All(
+            dict,
+            vol.Schema(
+                {
+                    PRESET_AWAY: vol.Coerce(float),
+                    PRESET_COMFORT: vol.Coerce(float),
+                    PRESET_ECO: vol.Coerce(float),
+                    PRESET_HOME: vol.Coerce(float),
+                    PRESET_SLEEP: vol.Coerce(float),
+                }
+            ),
+            cv.has_at_least_one_key(
+                PRESET_AWAY, PRESET_COMFORT, PRESET_ECO, PRESET_HOME, PRESET_SLEEP
+            ),
+        ),
```

Comment:
Why hardcode the allowed presets? 

There are no hardcoded presets in Home Assistant. Although some do offer translations when used and might connect with Google/Alexa.

See also https://developers.home-assistant.io/docs/core/entity/climate#presets

```suggestion
        vol.Optional(CONF_PRESETS): {str: vol.Coerce(float)},
```